agent/config/parse_windows.go

```diff
@@ -71,3 +89,24 @@ func isDomainJoined() (bool, error) {
 
 	return status == syscall.NetSetupDomainName, nil
 }
+
+// isWindows2016 is used to check if container instance is versioned Windows 2016
+// Reference: https://godoc.org/golang.org/x/sys/windows/registry
+func isWindows2016() bool {
+	key, err := registry.OpenKey(registry.LOCAL_MACHINE, `SOFTWARE\Microsoft\Windows NT\CurrentVersion`, registry.QUERY_VALUE)
+	if err != nil {
+		seelog.Errorf("Unable to open Windows registry key to determine Windows version: %v", err)
```

Comment:
If key read fails, should just return `false`.

Reply:
synced offline.